### PR TITLE
New features for unsteady aerodynamics modeling

### PR DESCRIPTION
**Pull request by bjonkman**
**Feature or improvement description**
This pull request includes the following new features:
- UnsteadyAero can be turned off based on a node's radial location (as opposed to setting c_nalpha = 0 in individual airfoil files). Two new lines in the AeroDyn input file allow the user to specify the radial locations where unsteady aero will be on.
- DBEMT_Mod = 3 is now available for time-marching simulations.
- UAMod = 4 (HGM) is now available for time-marching simulations.
- A new UAMod = 5 option is available. This model is based on HGM with one additional state.
- Airfoil tables can now be constant (i.e., have fewer than 3 lines). UA is turned off for constant airfoil tables.
- The UA parameters listed in the Airfoil files can now be calculated internally instead of user-specified. If any of the lines containing UA parameters are missing from the file, they get set by the code.

**Related issue, if one exists**
None

**Impacted areas of the software**
AeroDyn

**Additional supporting information**
- I also added some more validity checks on AFI UA parameters

**Test results, if applicable**
These changes do not change any existing test cases. All changes to the input file are optional.

### CODE REVIEW

**Comment by andrew-platt** [comment]
@rafmudaf and @ebranlard
Should we merge this into the RC3.1 branch instead of dev?  That way we can keep the current dev as the 3.0 release (hopefully releasing soon).

**Comment by ebranlard** [comment]
As a small note, I've merged this branch with f/driver into my branch f/driver-ua, where I've added two other dynamic stall models (Oye and Boeing-Vertol), but it's fine by me if we do this in two pull requests. (Mine will change the polar input file, adding the relative thickness). We might just be both doing the merge with dev..

**Review by andrew-platt — APPROVED** [review]
This looks good to me.

I really like the method of dealing with missing lines in input file parsing.  I'll have to use it in other codes.

File: docs/source/user/api_change.rst [hunk 1]
```diff
@@ -57,6 +67,8 @@ Added in OpenFAST v2.6.0
 Module                                        Line  Flag Name        Example Value
 ============================================= ==== =============== ========================================================================================================================================================================================================
 AeroDyn 15                                         TwrTi               0.0000000E+00  6.0000000E+00  1.0000000E+00  1.0000000E-01                 [additional column in *Tower Influence and Aerodynamics* table]
+AeroDyn 15                                    40   UAStartRad          0.25    UAStartRad          - Starting radius for dynamic stall (fraction of rotor radius) [used only when AFAeroMod=2; if line is missing UAStartRad=0]
```
Comment (andrew-platt) [inline docs/source/user/api_change.rst:70]: Was this change introduced in 2.6, or with this PR (in which case this and the next line should go with the section above).

Comment (andrew-platt) [reply docs/source/user/api_change.rst]: Figured this out.  This development branch existed prior to v2.6 and v3.0 releases.  Resolved with commit.

**Review by ebranlard — APPROVED** [review]
This is al good to me. 

I've started using this branch in my branch `f/driver-ua` and have not seen any issue. We might need to do small adjustments on UAMod 4 and 5 in the future, but nothing that would prevent the merge now.

**Comment by ebranlard** [comment]
I've added some documentation here: 

     https://ebranlard-openfast.readthedocs.io/en/f-driver-ua/source/user/aerodyn/theory_ua.html
     
Which will become available when my branch get's merged. I believe it's fine to wait a couple of months for that.

**Comment by bjonkman** [comment]
Since #729 was merged into #688, and #688 has subsequently been merged into OpenFAST/dev, the remaining changes here are minimal: some documentation of api changes, comments, and minor code cleanup.

**Pull request merged**